### DISCUSSION TITLE
Server / Forges / Use string type for PullRequest as ForgeRemoteID instead of int64

### DISCUSSION THREAD

**Discussion by mzampetakis** [woodpecker-ci/woodpecker#2746] [General]
Currently, for the Repository object the ForgeRemoteID has an underlying type of string. This makes completely sense as a forge might use some kind of int or some string (e.g. UUID) for the repository ID. The exact same approach is followed for the User object.

However, for the PullRequest Object, the ID is called `Index` (not `ForgeRemoteID`) and its type is `int64`. This might seems a slightly different approach compared to other entities which follow the previously mention approach. Currently there is no issue with the `int64` use as all supported forges use an integer as a Pull Request ID. 

Taking into consideration that some forges (existing or new) might support non-integer index for Pull Request but also for uniformity reasons I propose this to be refactored and match the rest of the types (`ForgeRemoteID`).

I am aware that this might need some refactor and I could work on this.
What do you think?

**Comment by qwerty287** [comment]
I see your point and I would be fine with this, so feel free to open a PR.

**Reply by mzampetakis** [reply]
This is great! Thanks!
I will work on this in the upcoming days!

**Comment by 6543** [comment]
https://github.com/woodpecker-ci/woodpecker/pull/2823 got merged
